src/script/event/EventRepositorySpec.ts

```diff
@@ -468,15 +491,19 @@ describe('EventRepository', () => {
 
     it('saves a text message with link preview with an ID previously used by the same user for a plain text message', () => {
       previously_stored_event = JSON.parse(JSON.stringify(event));
-      spyOn(testFactory.event_service, 'loadEvent').and.returnValue(Promise.resolve(previously_stored_event));
-      spyOn(testFactory.event_service, 'replaceEvent').and.returnValue(Promise.resolve(previously_stored_event));
+      jest
+        .spyOn(testFactory.event_service, 'loadEvent')
+        .mockImplementation(() => Promise.resolve(previously_stored_event));
```

Comment:
Suggestion: Can also be written as
```suggestion
        .mockImplementation(async () => previously_stored_event);
```